lighthouse/src/main.rs

```diff
@@ -372,21 +372,29 @@ fn run<E: EthSpec>(
     // Construct the path to the log file.
     let mut log_path: Option<PathBuf> = clap_utils::parse_optional(matches, "logfile")?;
     if log_path.is_none() {
-        log_path = match matches.subcommand_name() {
-            Some("beacon_node") => Some(
+        log_path = match matches.subcommand() {
+            ("beacon_node", _) => Some(
                 parse_path_or_default(matches, "datadir")?
                     .join(DEFAULT_BEACON_NODE_DIR)
                     .join("logs")
                     .join("beacon")
                     .with_extension("log"),
             ),
-            Some("validator_client") => Some(
-                parse_path_or_default(matches, "datadir")?
-                    .join(DEFAULT_VALIDATOR_DIR)
-                    .join("logs")
-                    .join("validator")
-                    .with_extension("log"),
-            ),
+            ("validator_client", Some(vc_matches)) => {
+                let base_path = if vc_matches.is_present("validators-dir") {
+                    parse_path_or_default(vc_matches, "validators-dir")?
+                } else {
+                    parse_path_or_default(matches, "datadir")?
+                };
+
+                Some(
+                    base_path
+                        .join(DEFAULT_VALIDATOR_DIR)
```

Comment:
I think this `join` should be in the `else` branch above so that we don't push an extra `validators` path segment onto the validators dir. E.g. if we pass `--validators-dir /var/lib/lighthouse/validators` the logs should go in `/var/lib/lighthouse/validators/logs/validator.log` rather than `/var/lib/lighthouse/validators/validators/logs/validator.log`

Reply:
Ooh good catch